src/net_processing.cpp

```diff
@@ -2752,22 +2752,24 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         bool fAnnounceUsingCMPCTBLOCK = false;
         uint64_t nCMPCTBLOCKVersion = 0;
         vRecv >> fAnnounceUsingCMPCTBLOCK >> nCMPCTBLOCKVersion;
-        if (nCMPCTBLOCKVersion == 1 || nCMPCTBLOCKVersion == 2) {
```

Comment:
Was there a reason for switching this to an early return, or is that now an artifact of old changes and rebases?

Just for readability, it's more clear imo wrapped in a `if` for the version check. I missed the early return when re-reviewing and was confused about why this change is safe.

Reply:
I think this is just personal taste. I _much_ prefer early returns where they're possible, especially compared with some of the very deeply nested if statements that we have in the code.

Reply:
This is nitty and not related to the code change, but I disagree with this. Deeply nested code, ugly as it may be, demonstrates the complexity of the logic in-place. Early returns don't simplify the logic, they just add sneaky trap-doors that aren't visible in diffs (In this case I misread the intent of the code because github didn't show enough context)